internal/app/singularity/remote_add.go

```diff
@@ -36,6 +35,12 @@ func RemoteAdd(configFile, name, uri string, global bool) (err error) {
 		return fmt.Errorf("while parsing remote config data: %s", err)
 	}
 
+	u, err := url.Parse(uri)
+	if err != nil {
+		return err
+	}
+	e := remote.EndPoint{URI: u.Host + u.Path}
```

Comment:
Though it might not be necessary, I would use `path.Join()` to concatenate the host and path since it will ensure it is properly formatted. Also we should set `System: global` when we create the endpoint